test/api-test.js

```diff
@@ -2608,6 +2608,115 @@ describe('RippleAPI', function () {
     assert.deepEqual(signature, responses.sign.signAs);
   });
 
+  it('sign - succeeds - prepared payment', async function () {
```

Comment:
I think there's a few code paths that aren't fully unit tested. Can you add tests for them or explain why they're infeasible.
- Test missing signature / signers param (sign.ts:58)
- Test the new call to `checkFee()` works correctly (sign.ts:88). 

Aside, it would be great to get code coverage reporting hooked up to a CI on this library to make this easy for reviewers to visualize. Are there blockers to doing so (If not, I'm happy to help out).

Reply:
It _should_ be impossible for the signature / signers param to be missing; the check is there "just in case".

The same goes for the new call to checkFee; the fee is already checked prior to that, and the decoded transaction must be equal to the original (aside from the fields added during signing). I'll remove this redundant check.

[Code coverage reporting](https://github.com/ripple/ripple-lib/blob/develop/.nycrc) and [CI](https://github.com/ripple/ripple-lib/blob/develop/circle.yml) have been set up for this library at various times in the past several years, but they are currently broken (or in need of updates). There are no blockers. Your help is welcome!

Reply:
Acknowledged. Lets sync offline about getting code coverage and CI set up.